scripts/ui/inventory/hotbar.gd

```diff
@@ -0,0 +1,104 @@
+extends UserInterface
+
+const SlotClass = preload("res://scripts/ui/inventory/slot.gd") # Preload the Slot script
+@onready var hotbar = $HotbarSlots # Reference to the HotbarSlots node
+@onready var slots = $HotbarSlots.get_children() #List of all child nodes of HotbarSlots
+@onready var active_item_label = $ActiveItemLabel # Reference to the ActiveItemLabel node
+@onready var tooltip = $ActiveItemLabel/Tooltip # Reference to the Tooltip node
+var t
+
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	#Sets up the Hotbar
+	PlayerInventory.active_item_updated.connect(self.update_active_item_label)
+	for i in range(slots.size()):
+		slots[i].gui_input.connect(slot_gui_input.bind(slots[i]))
+		PlayerInventory.active_item_updated.connect(slots[i].refresh_style)
+		slots[i].slot_index = i
+		slots[i].slot_type = SlotClass.SlotType.HOTBAR
+	initialize_hotbar()
+	update_active_item_label()
+	
+func tooltip_change():
+	# Toggle the visibility of the tooltip
+	active_item_label.visible = !active_item_label.visible
+
+func update_active_item_label():
+	tooltip.visible = true
+	if PlayerInventory.active_item_slot == null or PlayerInventory.active_item_slot > 7:
+		return
+	if slots[PlayerInventory.active_item_slot].item != null:
+		active_item_label.text = slots[PlayerInventory.active_item_slot].item.item_name
+		tooltip.visible = true
+		var info_slots = get_tree().get_nodes_in_group("hotbar_info_slots")
+		info_slots[0].text = JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["ItemCategory"]
+		info_slots[1].text = slots[PlayerInventory.active_item_slot].item.item_name
+		info_slots[2].text = JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["Description"]
+		if JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["ItemCategory"] == "Consumable":
+			info_slots[3].text = "Adds " + str(JsonData.item_data[slots[PlayerInventory.active_item_slot].item.item_name]["AddHealth"]) + " Health"
```

Comment:
This whole system is incredibly inflexible. I apologize if I seem rude while writing these but this will cause problems later. If you want items like this, there are a variety of better ways than hard coding a bunch of cases. If you do want to do such hard coding, try moving it to a static script/autoload that can delegate function calling from there. That way, we can build on this later instead of restricting ourselves to a few options. Also another point where resources would save a ton of time.